treerec/tskit/tables.h

```diff
@@ -103,6 +102,16 @@ typedef struct {
     tsk_size_t nodes_length;
 } tsk_individual_t;
 
+//Moved from tables.c(sudharshan)
+typedef struct {
+    double left;
+    double right;
+    tsk_id_t parent;
+    tsk_id_t child;
+    double time;
+} edge_sort_t;
```

Comment:
I'm kinda scratching my head as to what this struct even is.  Why does the comparator for sorting the edge table use a special struct instead of just using the struct that the edge table actually uses?  I know this is tskit's design, not yours.  Just seems a bit odd.

Reply:
@petrelharp, do you understand why tskit does this?

Reply:
Yes - `time` is not stored by an `edge` (as seen [here](https://github.com/tskit-dev/tskit/blob/a82bab156996b0728c0021a430ee8001712573c1/c/tskit/tables.h#L139)); but we want to sort edges by time (I forget if this is parent or child time).